docs/_examples/vcenter-events.md

```diff
@@ -0,0 +1,223 @@
+---
+layout: default
+title:  Harden VMs in vSphere
+---
+
+# Introduction
+
+This tutorial will walk you through executing functions in response to vSphere events.
+_Note:_ Ensure you have followed the [Quickstart]({{ '/documentation/front/quickstart' | relative_url }}) guide and have a working Dispatch Virtual Appliance.
+
+Our goal is to execute a PowerShell function called hardenVM, which will apply additional hardening configuration on every virtual machine created in vSphere.
+
+# Create Seed Images
+
+Dispatch is bundled with a set of seed images for multiple languages to get you started easily. If you followed the guide, you should
+have a set of images created in your Dispatch VM. Execute the following command to check:
+
+```bash
+$ dispatch get images
+     NAME    |   URL                |    BASEIMAGE    | STATUS | CREATED DATE
+-----------------------------------------------------------------------------
+  java       | dispatch/f23f029e... | java-base       | READY  | ...
+  nodejs     | dispatch/6f04f67d... | nodejs-base     | READY  | ...
+  powershell | dispatch/edcbdda8... | powershell-base | READY  | ...
+  python3    | dispatch/1937b329... | python3-base    | READY  | ...
+```
+
+If the list is empty, you can create the seed images using the following command:
+
+```bash
+$ dispatch create seed-images
+Created BaseImage: nodejs-base
+Created BaseImage: python3-base
+Created BaseImage: powershell-base
+Created BaseImage: java-base
+Created Image: nodejs
+Created Image: python3
+Created Image: powershell
+Created Image: java
+```
+
+**Note**: You need to wait a short while for the images to be in the `READY` state. You can always check the status using the `dispatch get images` command.
+
+# Create a PowerShell image with PowerCLI dependency
+
+Our function needs a library that knows how to talk to vSphere. We will use `PowerCLI` SDK to do that. To bring the dependency for our function, we will prepare a dedicated image.
+
+1. Create a file named `deps.ps1` with PowerCLI dependency:
+
+```bash
+cat << EOF > deps.psd1
+@{
+  'VMware.PowerCLI' = 'latest'
+}
+EOF
+```
+
+2. create a new image using PowerShell base image:
+
+```
+$ dispatch create image powershell-powercli powershell-base --runtime-deps deps.psd1
+Created image: powershell-powercli
+```
+
+# Create a secret with vSphere credentials
+
+Our function also needs to know how to connect to vSphere, and what credentials to use. In order to do that, we will create a Dispatch secret.
+
+1. Create a JSON file with secret contents:
+
+```bash
+cat << EOF > vsphere.json
+{
+  "host": "<vCenter host URL, e.g. https://vcenter.example.com>",
+  "username": "<username>",
+  "password": "<password>"
+  "vcenterurl": "<full vCenter URL, e.g. username:password@vcenter.example.com:443>"
```

Comment:
Do we still have the password issues?  Also, at some point we should probably not require the URL since we already have all of the pieces which make up the URL.